packages/mom5/package.py

```diff
@@ -5,114 +5,177 @@
     version("access-esm1.5", branch="access-esm1.5")
     version("access-esm1.6", branch="master")
 
-    variant("restart_repro", default=True, description="Reproducible restart build.")
-
-    with when("@access-esm1.6,access-om2,legacy-access-om2-bgc"):
-        variant("deterministic",
-                default=False,
-                description="Deterministic build.")
-        variant("optimisation_report",
-                default=False,
-                description="Generate optimisation reports.")
+    # NOTE: @mom matches both mom_solo and mom_sis
+    build_system(
+        conditional("makefile", when="@access-om2,legacy-access-om2-bgc,access-esm1.5"),
+        conditional("cmake", when="@mom,access-om2,legacy-access-om2-bgc,access-esm1.6"),
+        default="cmake",
+    )
+
+    with when("build_system=cmake"):
+        variant("build_type", default="RelWithDebInfo",
+            description="CMake build type",
+            values=("Debug", "Release", "RelWithDebInfo")
+        )
+        variant("deterministic", default=False, description="Deterministic build")
+
+    with when("build_system=makefile"):
+        variant("restart_repro", default=True, description="Reproducible restart build.")
+        variant(
+            "deterministic",
+            default=False,
+            description="Deterministic build",
+            when="@access-om2,legacy-access-om2-bgc,access-esm1.6"
+        )
+        variant(
+            "optimisation_report",
+            default=False,
+            description="Generate optimisation reports",
+            when="@access-om2,legacy-access-om2-bgc,access-esm1.6"
+        )
 
-    with when("@access-om2,legacy-access-om2-bgc"):
+    with when("@mom,access-om2,legacy-access-om2-bgc,access-esm1.6"):
         depends_on("netcdf-c@4.7.4:")
         depends_on("netcdf-fortran@4.5.2:")
-        # Depend on virtual package "mpi".
         depends_on("mpi")
-        depends_on("datetime-fortran")
+
+    with when("@access-om2,legacy-access-om2-bgc,access-esm1.6"):
         depends_on("oasis3-mct+deterministic", when="+deterministic")
         depends_on("oasis3-mct~deterministic", when="~deterministic")
+
+    with when("@access-om2,legacy-access-om2-bgc"):
+        depends_on("datetime-fortran")
         depends_on("libaccessom2+deterministic", when="+deterministic")
         depends_on("libaccessom2~deterministic", when="~deterministic")
 
     # NOTE: Spack will also match "access-om2-legacy-bgc" here, that's why
     #       it has been renamed to "legacy-access-om2-bgc".
-    with when("@access-om2"):
+    with when("@access-om2,access-esm1.6"):
         depends_on("access-fms")
         depends_on("access-generic-tracers")
 
-    # access-esm1.5 and access-esm1.6
-    with when("@access-esm1.5:access-esm1.6"):
+    with when("@legacy-access-om2-bgc"):
+        depends_on("access-fms", when="build_system=cmake")
+        depends_on("access-generic-tracers", when="build_system=cmake")
```

Comment:
Consider adding a comment on why this has been introduced.

Reply:
Fixed in [45a1981](https://github.com/ACCESS-NRI/spack-packages/pull/238/commits/45a19817aee613637f25091560d4fa76b49a8623)